Sources/MapboxMaps/Location/Puck/Puck2D.swift

```diff
@@ -215,11 +220,13 @@ internal final class Puck2D: Puck {
             if configuration.showsAccuracyRing {
                 layerProperties[LocationIndicatorLayer.PaintCodingKeys.accuracyRadius.rawValue] = location.horizontalAccuracy
             }
-            switch puckBearingSource {
-            case .heading:
-                layerProperties[LocationIndicatorLayer.PaintCodingKeys.bearing.rawValue] = location.heading ?? 0
-            case .course:
-                layerProperties[LocationIndicatorLayer.PaintCodingKeys.bearing.rawValue] = location.course
+            if puckBearingEnabled {
+                switch puckBearingSource {
+                case .heading:
+                    layerProperties[LocationIndicatorLayer.PaintCodingKeys.bearing.rawValue] = location.heading ?? 0
+                case .course:
+                    layerProperties[LocationIndicatorLayer.PaintCodingKeys.bearing.rawValue] = location.course ?? 0
```

Comment:
We should have tests that fail if any of the lines like this don't have the `?? 0`